### ISSUE TITLE
Invalid rules for nested types and protocols

### ISSUE THREAD

**Issue by drochetti** [outfoxx/swiftpoet#60]
SwiftPoet implementation checks if the TypeSpec is a protocol when adding a nested type, however that rule is not 100% correct.

Here is a few scenarios that the current code can't handle correctly:

**Scenario A: The nested type is a typealias**

Protocols are allowed to have type aliases as nested types. Since `TypeAliasSpec` is a subclass of `AnyTypeSpec` the current implementation doesn't allow that. The following is a valid construct in Swift that cannot be built with SwiftPoet:

```swift
protocol MyProtocol {

  typealias NestedTypeAlias = String

}
```

**Scenario B: Protocol is a nested type of a struct/class/enum**

Protocols can't have nested types (other than type aliases) and cannot in any scenario be a nested type of another construct. The Swift compiler in such scenarios is `Protocol 'Nested' cannot be nested inside another declaration`. Therefore, the following code is possible to be built using SwiftPoet but it's invalid:

```swift
struct MyStruct {

  protocol MyNestedProtocol {
  }

}
```

**Comment by drochetti** [comment]
I opened a PR to fix the issue above. Let me know if you have any feedback. I'd love to get this merged since I'm writing some code that use type aliases.

**Comment by kdubb** [comment]
Fixed via #61

**Comment by kdubb** [comment]
@drochetti Thanks for reporting and thanks for the great PR that fixed this issue!
